platform-sdk/docs/core/tipset-algorithm.md

```diff
@@ -0,0 +1,260 @@
+# Introduction
```

Comment:
This document covers the idea of the algorithm, but nothing about how it is implemented in the code. The code adds a new queue thread. This documentation should include how it fits into the pipeline and what happens when it fills and blocks on add.

Reply:
I'd actually prefer to avoid going into implementation in this document. The goal of this document was to present a high level conceptual definition of the tipset algorithm.

Reply:
Is there a different document then that talks about the design/impl?  (today's deadlock occurrence indicates that there is complexity in the queuing of events and things that should be explained)

Reply:
We are currently working on documentation for this. Good diagrams are the best way to convey this information, in my opinion. That change will merge in a different PR though, the intention of this one was to document the theory behind the tipset algorithm.